### PR TITLE
Improves support for SM4s

### LINKED ISSUE #330: Sub-second precision in SM4
I am using Emu to extract metadata from SM4 recordings. Sub-second precision in the timestamp is causing problems. 
image
image

### PR DESCRIPTION
**Pull request by atruskie**
- Adds high precision timer support for SM4s. Fixes #330
- Adds support for microphone sensitivity reporting for Sm4s partially addressed #331
- Adds make, model, and sensor name for SM4. Partially addresses #296

- BREAKING change: resets channel identifier to match industry standard. i.e. the first channel is 0 (left). We were counting from 1 which is not the most popular format: https://en.wikipedia.org/wiki/Surround_sound#Channel_notation
- BREAKING: Fixed a bug with the Location parsing code! Coordinates from WA sensors will now report the correct hemisphere

### CODE REVIEW

File: src/Emu/Meta.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Emu
```
Comment (hudson-newey) [inline src/Emu/Meta.cs:1]:
```suggestion
// <copyright file="Meta.cs" company="QutEcoacoustics">
// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
// </copyright>
﻿namespace Emu
```

File: src/Emu/Metadata/WildlifeAcoustics/WamdExtractor.cs [hunk 2]
```diff
@@ -31,46 +32,72 @@ public ValueTask<Recording> ProcessFileAsync(TargetInformation information, Reco
...
+                    location = wamdData.GpsFirst;
+                }
+                else if (wamdData.PosLast is not null)
+                {
+                    location = wamdData.PosLast;
+                }
+
+                if (wamdData.FileStartTime is not null)
+                {
+                    if (wamdData.FileStartTime.Value.Case is OffsetDateTime o)
```
Comment (hudson-newey) [inline src/Emu/Metadata/WildlifeAcoustics/WamdExtractor.cs:55]: Possible rename `o` to a more descriptive name?

I had to read the proceeding lines of code to figure out how `o` was used

**Review by hudson-newey — APPROVED** [review]
LGTM, just some simple file header errors caught in CI & a variable name

File: src/Emu/Metadata/WildlifeAcoustics/WamdExtractor.cs (hunk 2 above)
Comment (atruskie) [reply src/Emu/Metadata/WildlifeAcoustics/WamdExtractor.cs:55]: hmmm this is a common convention for temporary variables that created just due to a cast. Here `o` it is just the cast value of `wamdData.FileStartTime`. However, any example of something difficult to read takes priority

**Review by hudson-newey — APPROVED** [review]

**Comment by elisagirola** [comment]
Hi Anthony, have you already released this change?

**Comment by atruskie** [comment]
Not yet, I'm doing one more feature and then I'll release.

It was way harder than expected but it should be worth it

**Comment by elisagirola** [comment]
Awesome. Thanks Anthony. Much appreciated.

**Pull request merged**